src/probnum/filtsmooth/gaussian/_kalman.py

```diff
@@ -165,30 +42,20 @@ def filtsmooth(
         --------
         TimeSeriesRegressionProblem: a regression problem data class
         """
-        filter_result = self.filter(
-            regression_problem,
-            _previous_posterior=_previous_posterior,
-        )
+        filter_result = self.filter(*args, **kwargs)
         filter_posterior, info_dicts = filter_result
         smooth_posterior = self.smooth(filter_posterior)
         return smooth_posterior, info_dicts
 
-    def filter(
-        self,
-        regression_problem: problems.TimeSeriesRegressionProblem,
-        _previous_posterior: Optional[_timeseriesposterior.TimeSeriesPosterior] = None,
-    ):
+    def filter(self, *args, **kwargs):
```

Comment:
This wrapper around the state generator works with args and kwargs for the following reason:

Soon, the regression problems might change types (discrete vs continuous), and the filter() and iterated_filtsmooth_generator() signature adapts accordingly. But this function here (and the filtsmooth() style methods below) remains the same. args and kwargs lead to a smaller, more meaningful diff in the future.